src/WebJobs.Script.WebHost/Models/RunFromPackageContext.cs

```diff
@@ -31,14 +26,19 @@ public RunFromPackageContext(string envVarName, string url, long? packageContent
 
         public bool IsScmRunFromPackage()
         {
-            return string.Equals(EnvironmentVariableName, EnvironmentSettingNames.ScmRunFromPackage,
-                        StringComparison.OrdinalIgnoreCase);
+            return string.Equals(EnvironmentVariableName, EnvironmentSettingNames.ScmRunFromPackage, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public bool IsRunFromPackage(ScriptApplicationHostOptions options, ILogger logger)
+        {
+            return (IsScmRunFromPackage() && ScmRunFromPackageBlobExists(options, logger)) || (!IsScmRunFromPackage() && !string.IsNullOrEmpty(Url) && Url != "1");
         }
 
-        public async Task<bool> IsRunFromPackage(ILogger logger)
+        private bool ScmRunFromPackageBlobExists(ScriptApplicationHostOptions options, ILogger logger)
         {
-            return (IsScmRunFromPackage() && await _runFromPackageCloudBlockBlobService.BlobExists(Url, EnvironmentVariableName, logger)) ||
-                   (!IsScmRunFromPackage() && !string.IsNullOrEmpty(Url) && Url != "1");
+            var blobExists = options.ScmRunFromPackageBlobExists;
+            logger.LogDebug($"Checking if {EnvironmentSettingNames.ScmRunFromPackage} points to an existing blob: {blobExists}");
```

Comment:
nit: I wonder if we shouldn't have this log where we're actually doing the existence check, since it seems to imply we're doing it at this point when we're not.

Reply:
I agree that the message is misleading and could be improved. I decided to log the value of blob existence here because adding the logger to `ScriptApplicationHostOptionsSetup` wasn't an option; it introduced recursive dependencies.

Let me refactor this class by simplifying the checks like you're suggesting in the comment above and improve the log message as well.

Reply:
I've updated the message to omit the "Checking if" part. Please let me know if you have a better idea of where/how we can log when we're checking if the blob exists.